ios/wrappers/push/CioRctPushMessaging.swift

```diff
@@ -0,0 +1,117 @@
+import Foundation
+import CioInternalCommon
+import CioMessagingPush
+
+enum PushPermissionStatus: String, CaseIterable {
+    case denied
+    case notDetermined
+    case granted
+    
+    var value: String {
+        return rawValue.firstUppercased
+    }
+}
+
+@objc(CioRctPushMessaging)
+class CioRctPushMessaging: NSObject {
+
+    @objc static func requiresMainQueueSetup() -> Bool {
+        false /// false because our native module's initialization does not require access to UIKit
+    }
+    
+    // Tracks `opened` push metrics when a push notification is interacted with.
+    @objc
+    func trackNotificationResponseReceived(payload: NSDictionary) {
+        trackPushMetrics(payload: payload, event: .opened)
+    }
+    
+    // Tracks `delivered` push metrics when a push notification is received.
+    @objc
+    func trackNotificationReceived(payload: NSDictionary) {
+        
+        trackPushMetrics(payload: payload, event: .delivered)
+    }
+   
+    // Get the currently registered device token for the app
+    @objc
+    func getRegisteredDeviceToken(resolver resolve: @escaping(RCTPromiseResolveBlock), rejecter reject: @escaping(RCTPromiseRejectBlock)) -> Void {
+        
+         guard let token = CustomerIO.shared.registeredDeviceToken else {
+            reject(CustomerioConstants.cioTag, CustomerioConstants.showDeviceTokenFailureError, nil)
+             return
+        }
+        resolve(token)
+    }
+
+    private func trackPushMetrics(payload: NSDictionary, event : Metric) {
+        guard let deliveryId = payload[CustomerioConstants.CioDeliveryId] as? String, let deviceToken = payload[CustomerioConstants.CioDeliveryToken] as? String else
+        {return}
+        
+        MessagingPush.shared.trackMetric(deliveryID: deliveryId, event: event, deviceToken: deviceToken)
+    }
+    
+    @objc(showPromptForPushNotifications:resolver:rejecter:)
+    func showPromptForPushNotifications(options : Dictionary<String, AnyHashable>, resolver resolve: @escaping(RCTPromiseResolveBlock),  rejecter reject: @escaping(RCTPromiseRejectBlock)) -> Void {
+        
+        // Show prompt if status is not determined
+        getPushNotificationPermissionStatus { status in
+            if status == .notDetermined {
+                self.requestPushAuthorization(options: options) { permissionStatus in
+                    
+                    guard let status = permissionStatus as? Bool else {
+                        reject(CustomerioConstants.cioTag, CustomerioConstants.showPromptFailureError, permissionStatus as? Error)
+                        return
+                    }
+                    resolve(status ? PushPermissionStatus.granted.value : PushPermissionStatus.denied.value)
+                }
```

Comment:
```suggestion
self.requestPushAuthorization(options: options) { permissionStatus in
                    guard let isGranted = permissionStatus as? Bool else {
                        reject(CustomerioConstants.cioTag, CustomerioConstants.showPromptFailureError, permissionStatus as? Error)
                        return
                    }
                    resolve(isGranted ? PushPermissionStatus.granted.value : PushPermissionStatus.denied.value)
                }
```

maybe lets use a local variable instead of reassigning status multiple times, because it can cause confusion as to which variable was intended to change vs not.

Reply:
I don’t have a strong preference for the variable name, but since it’s within a local scope in the guard let statement, it might make sense to use it as is. I have updated the variable name to `isGranted` in case it caused confusion.